blocks/graph.py

```diff
@@ -37,6 +37,7 @@ def _get_variables(self):
         self.applies = set()
         self.application_calls = set()
         self.updates = []
+        self.auxiliary_variables = set()
```

Comment:
Ah, we can do that too, but auxiliary variables don't have a role right now; they're auxiliary variables simply because they were attached to an application call. But yeah, we need an easy way to e.g. return auxiliary variables which have a `VariableRole.COST` role.